### PR TITLE
[Android] Fix 'Unused assignment' inspection.

### PR DESCRIPTION
**Pull request by adamradocz**
**Description of the Change**
_Unused assignment_
This inspection points out the cases where a variable value is never used after its assignment, i.e.:  - the variable never gets read after assignment OR  - the value is always overwritten with another assignment before the next variable read OR  - the variable initializer is redundant (for one of the above two reasons)

**Release Notes**
N/A

### CODE REVIEW

**Comment by brevilo** [comment]
Good point. Let's not waste memory...

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientInterfaceImplementation.java [hunk 1]
```diff
@@ -362,8 +361,7 @@ public Boolean synchronizeAcctMgr(String url) {
     	} else {if(Logging.DEBUG) Log.d(Logging.TAG,"rpc.getProjectConfig returned false.");}
 		
     // 2nd acct_mgr_rpc with <use_config_file/>
-		success = false;
-		AcctMgrRPCReply reply2 = null;
+        AcctMgrRPCReply reply2;
```
[the remark this answers is not in the document]
Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientInterfaceImplementation.java]: Fixed.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientInterfaceImplementation.java [hunk 2]
```diff
@@ -335,9 +335,8 @@ public AcctMgrRPCReply addAcctMgr(String url, String userName, String pwd) {
 	public Boolean synchronizeAcctMgr(String url) {
 
 	// 1st get_project_config for account manager url
-		Boolean success = false;
-		ProjectConfig reply = null;
-		success = getProjectConfig(url);
+		Boolean success = getProjectConfig(url);
+		ProjectConfig reply;
```
[the remark this answers is not in the document]
Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientInterfaceImplementation.java]: Fixed.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.java [hunk 3]
```diff
@@ -605,9 +605,7 @@ private Boolean runClient() {
      * @return Boolean success
      */
 	private Boolean connectClient() {
-		Boolean success = false;
-		
-        success = clientInterface.open(clientSocketAddress);
+		Boolean success = clientInterface.open(clientSocketAddress);
```
[the remark this answers is not in the document]
Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.java]: Fixed.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 4]
```diff
@@ -61,7 +61,7 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
 	
 	@SuppressLint("DefaultLocale")
 	public static String formatSize(double fBytesSent, double fFileSize) {
-		String buf = "";
+		String buf;
```
[the remark this answers is not in the document]
Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: Fixed.

**Comment by adamradocz** [comment]
Done.
The problem is the project is a mess in terms of indentation. Almost every file have different indentation style (tab and space), and fairly often the two styles mixed within 1 line.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientStatus.java [hunk 5]
```diff
@@ -490,7 +490,7 @@ public String getCurrentStatusDescription() {
 					statusString = ctx.getString(R.string.suspend_battery_overheating);
 					break;
 				case BOINCDefs.SUSPEND_REASON_USER_ACTIVE:
-					Boolean suspendDueToScreenOn = false;
+					Boolean suspendDueToScreenOn;
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientStatus.java]: Could this and next line be combined into one?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 6]
```diff
@@ -61,7 +61,7 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
 	
 	@SuppressLint("DefaultLocale")
 	public static String formatSize(double fBytesSent, double fFileSize) {
-		String buf = "";
+	    String buf;
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: Could we just get rid of this variable at all? Then next if-else statements could be simplified.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientStatus.java (hunk 5 above)
Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientStatus.java]: Done.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java (hunk 6 above)
Comment (adamradocz) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: Done.

**Comment by JuhaSointusalo** [comment]
> The problem is the project is a mess in terms of indentation. Almost every file have different indentation style (tab and space), and fairly often the two styles mixed within 1 line.

Pause opening new pull requests.
Wait for the currently open ones to be reviewed and merged.
Then open a new pull request that sorts out the whitespace issues in Android code once and for all.
Once that whitespace pull request is merged then resume with the code clean up.

Resist adding anything other than whitespace changes to the whitespace pull requests. That makes reviewing it very easy. Android studio probably has some tool to assist in code formatting. Just make sure you have configured all formatting settings, not only space vs tabs.

Also make the whitespace changes only to Android code. Right now you are the only one making changes there but we have too many open pull requests in other parts and whitespace changes will conflict with those pull requests.

**Comment by adamradocz** [comment]
@JuhaSointusalo Thanks, I'm going to do that.

**Comment by adamradocz** [comment]
Which one? I can't see any indentation problem in the new code.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 7]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
-	        	buf = String.format("%.2f/%.2f MB", fBytesSent/xMega, fFileSize/xMega);
-	        } else if (fFileSize >= xKilo) {
-	        	buf = String.format("%.2f/%.2f KB", fBytesSent/xKilo, fFileSize/xKilo);
-	        } else {
-	        	buf = String.format("%.0f/%.0f bytes", fBytesSent, fFileSize);
+	        if (fFileSize >= xTera) {
+	            return String.format("%.2f/%.2f TB", fBytesSent/xTera, fFileSize/xTera);
 	        }
+	        if (fFileSize >= xGiga) {
+	        	return String.format("%.2f/%.2f GB", fBytesSent/xGiga, fFileSize/xGiga);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: Here

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 8]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
+	        }
+	        if (fFileSize >= xMega) {
+				return String.format("%.2f/%.2f MB", fBytesSent/xMega, fFileSize/xMega);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: And here

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 9]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
+	        }
+	        if (fFileSize >= xKilo) {
+				return String.format("%.2f/%.2f KB", fBytesSent/xKilo, fFileSize/xKilo);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: And here

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 10]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
-	        } else if (fBytesSent >= xGiga) {
-	        	buf = String.format("%.2f GB", fBytesSent/xGiga);
-	        } else if (fBytesSent >= xMega) {
-	        	buf = String.format("%.2f MB", fBytesSent/xMega);
-	        } else if (fBytesSent >= xKilo) {
-	        	buf = String.format("%.2f KB", fBytesSent/xKilo);
-	        } else {
-	        	buf = String.format("%.0f bytes", fBytesSent);
+	        if (fBytesSent >= xTera) {
+				return String.format("%.2f TB", fBytesSent/xTera);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: And here

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 11]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
 	        }
+	        if (fBytesSent >= xGiga) {
+				return String.format("%.2f GB", fBytesSent/xGiga);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: And here

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 12]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
+	        }
+	        if (fBytesSent >= xMega) {
+				return String.format("%.2f MB", fBytesSent/xMega);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: And here

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java [hunk 13]
```diff
@@ -61,39 +61,41 @@ public static String translateNetworkSuspendReason(Context ctx, int reason) {
...
+	        }
+	        if (fBytesSent >= xKilo) {
+				return String.format("%.2f KB", fBytesSent/xKilo);
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/utils/BOINCUtils.java]: And here

**Comment by brevilo** [comment]
> Done.
> The problem is the project is a mess in terms of indentation. Almost every file have different indentation style (tab and space), and fairly often the two styles mixed within 1 line.

I agree. Apart from the indentation-style-detection in your editor I recommend you turn on *invisibles* for the time being. This way you'll see what's used (if your editor isn't able to auto-detect) and lets you adjust right away.

**Comment by brevilo** [comment]
> Resist adding anything other than whitespace changes to the whitespace pull requests. That makes reviewing it very easy.

Couldn't agree more 👍

**Comment by AenBleidd** [comment]
@adamradocz, could you please resolve merge conflicts one more time? Thanks

**Comment by AenBleidd** [comment]
Looks fine. Thanks. Merging...

**Pull request merged**